### PR TITLE
Potential fix for code scanning alert no. 3: Reflected server-side cross-site scripting

### PR DESCRIPTION
**Pull request by gmorador-tribu**
Potential fix for [https://github.com/hypothesis/bouncer/security/code-scanning/3](https://github.com/hypothesis/bouncer/security/code-scanning/3)

To fix this without changing behavior, normalize/encode `group` before inserting it into `fragment`, the same way `q` is handled. The best minimal fix is to percent-encode `group` using `urllib.parse.quote(..., safe="")` when reading it from `request.params`. This preserves intended semantics (group still passed in fragment) while preventing injection payloads from being reflected as active markup/script.

In `bouncer/views.py`, inside `goto_url`:
- Change the `group` assignment (currently raw `request.params.get("group", "")`) to a quoted value.
- Keep existing logic unchanged otherwise.

No new imports are required because `parse` is already imported from `urllib`.

_Suggested fixes powered by Copilot Autofix. Review carefully before merging._

### CODE REVIEW

File: bouncer/views.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def goto_url(request):
     # append our own.
     url = parse.urldefrag(url)[0]
 
-    group = request.params.get("group", "")
+    group = parse.quote(request.params.get("group", ""), safe="")
     query = parse.quote(request.params.get("q", ""))
```
[the remark this answers is not in the document]
Comment (gmorador-tribu) [reply bouncer/views.py:170]: @copilot also add the unit test for edge cases
